lib/kb_plant_rast/kb_plant_rastImpl.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+#BEGIN_HEADER
+import os,uuid
+from KBaseReport.KBaseReportClient import KBaseReport
+from Workspace.WorkspaceClient import Workspace as workspaceService
+#END_HEADER
+
+
+class kb_plant_rast:
+    '''
+    Module Name:
+    kb_plant_rast
+
+    Module Description:
+    A KBase module: kb_plant_rast
+    '''
+
+    ######## WARNING FOR GEVENT USERS ####### noqa
+    # Since asynchronous IO can lead to methods - even the same method -
+    # interrupting each other, you must be *very* careful when using global
+    # state. A method could easily clobber the state set by another while
+    # the latter method is running.
+    ######################################### noqa
+    VERSION = "0.0.1"
+    GIT_URL = "https://github.com/kbaseapps/kb_plant_rast"
+    GIT_COMMIT_HASH = "a652c0120abf90e97d0f0214f8ed4174f27b9a09"
+
+    #BEGIN_CLASS_HEADER
+    #END_CLASS_HEADER
+
+    # config contains contents of config file in a hash or None if it couldn't
+    # be found
+    def __init__(self, config):
+        #BEGIN_CONSTRUCTOR
+        self.workspaceURL = config['workspace-url']
+        self.scratch = os.path.abspath(config['scratch'])
+        self.callback_url = os.environ['SDK_CALLBACK_URL']
+        #END_CONSTRUCTOR
+        pass
+
+
+    def annotate_plant_transcripts(self, ctx, input):
+        """
+        :param input: instance of type "AnnotatePlantTranscriptsParams" ->
+           structure: parameter "input_ws" of String, parameter
+           "input_genome" of String, parameter "output_genome" of String
+        :returns: instance of type "AnnotatePlantTranscriptsResults" ->
+           structure: parameter "report_name" of String, parameter
+           "report_ref" of String
+        """
+        # ctx is the context object
+        # return variables are: output
+        #BEGIN annotate_plant_transcripts
+
+        token = ctx['token']
+        wsClient = workspaceService(self.workspaceURL, token=token)
+
+        # Retrieve plant genome
+        plant_genome = wsClient.get_objects2({'objects': [{'ref' : input['input_ws']+'/'+input['input_genome']}]})['data'][0]
```

Comment:
You should use DataFileUtil so that provenance is automatically correctly handled: https://github.com/kbaseapps/DataFileUtil/blob/master/DataFileUtil.spec#L452